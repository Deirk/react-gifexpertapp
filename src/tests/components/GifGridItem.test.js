import { shallow } from 'enzyme';
import React from 'react';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en GifGridItem.js', () => {
    
    const title = "Titulo de prueba";
    const url = "url/de/prueba";

    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test('debe mostrar correctamente el componente en el snapshot', () => {


        expect(wrapper).toMatchSnapshot();    
    
    });

    test('debe de tener un parrafo con el title ', () => {
        
        const p = wrapper.find('p');

        expect(p.text().trim()).toBe(title);

    })
    
    test('debe de tener la imagen igual al url y alt de los props', () => {
        
        const img = wrapper.find('img');

        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);

    })
    
    test('debe tener animate__fadeIn', () => {
        
        const div = wrapper.find('div');

        expect(div.prop('className')).toContain('animate__fadeIn');

    })
    
    

});
