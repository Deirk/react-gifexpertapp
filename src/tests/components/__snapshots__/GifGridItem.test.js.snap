// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem.js debe mostrar correctamente el componente en el snapshot 1`] = `
<div
  className="card animate__fadeIn"
>
  <img
    alt="Titulo de prueba"
    src="url/de/prueba"
  />
  <p>
     
    Titulo de prueba
     
  </p>
</div>
`;
